trade-finance-manager-ui/server/controllers/case/amendments/amendmentBankReviewDate.controller.js

```diff
@@ -22,7 +22,9 @@ const getAmendmentBankReviewDate = async (req, res) => {
     return res.redirect('/not-found');
   }
 
-  if (!changeCoverEndDate || !isTfmFacilityEndDateFeatureFlagEnabled()) {
+  const isFacilityEndDateEnabled = isTfmFacilityEndDateFeatureFlagEnabled() && facility.facilitySnapshot.isGef;
```

Comment:
Should we instead have an e.g `endDateEnabled` flag coming from the `mapGefFacility` API mapping? Then the UI can just consume with `if (facility.endDateEnabled) { ...`

Same logic is repeated twice in this file at least

Reply:
the feature flag is independant of the facility so belongs off the object -- we had extracted this out to a helper to avoid repetition.

in reality, it won't matter in the long run as the FF will be removed

Reply:
Thanks @AlexBramhill - not a blocker for this PR; When it comes to removing the feature flag, will it just be a case of searching for and removing  `isFacilityEndDateEnabled` and `isTfmFacilityEndDateFeatureFlagEnabled`?

Would be nice to make sure all such checks have the exact same name for removal; Otherwise maybe we could/should create a new "remove X FF" ticket and reference the ticket number

Reply:
`isTfmFacilityEndDateFeatureFlagEnabled` comes from `is-feature-flag-enabled` which is a way of us validating and parsing feature flag values (very much like our config files).

So when it comes to removing we just need to remove `isTfmFacilityEndDateFeatureFlagEnabled` -- the reason this isn't done in line in some places and declares a variable like here is as we do not know when this is going to be removed so it's worth keeping it readable